### PR TITLE
LLM: add eval func

### PR DESCRIPTION
**Pull request by WeiguangHan**
@hkvision Add eval func for models.

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 1]
```diff
@@ -344,16 +344,19 @@ def run_transformer_int4_gpu(repo_id,
...
     elif repo_id in LLAMA_IDS:
         model = AutoModelForCausalLM.from_pretrained(model_path, load_in_low_bit=low_bit, trust_remote_code=True,
                                                      use_cache=True)
+        model=model.eval()
         tokenizer = LlamaTokenizer.from_pretrained(model_path, trust_remote_code=True)
         model = model.to('xpu')
     else:
         model = AutoModelForCausalLM.from_pretrained(model_path, optimize_model=True, load_in_low_bit=low_bit,
                                                      trust_remote_code=True, use_cache=True)
+        model=model.eval()
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:359]: Can we add this in Line 365, one time for all if branches?

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 2]
```diff
@@ -275,17 +275,16 @@ def run_optimize_model(repo_id,
...
         model = AutoModelForCausalLM.from_pretrained(model_path, trust_remote_code=True,
                                                      use_cache=True, low_cpu_mem_usage=True)
-        model = optimize_model(model, low_bit=low_bit)
         tokenizer = LlamaTokenizer.from_pretrained(model_path, trust_remote_code=True)
     else:
         model = AutoModelForCausalLM.from_pretrained(model_path, torch_dtype='auto', low_cpu_mem_usage=True)
-        model = optimize_model(model, low_bit=low_bit)
         tokenizer = AutoTokenizer.from_pretrained(model_path)
+    model = optimize_model(model, low_bit=low_bit)
+    model=model.eval()
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:287]: space before and after =

Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py:287]: Sure.

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 3]
```diff
@@ -425,24 +425,22 @@ def run_optimize_model_gpu(repo_id,
...
-        if isinstance(model, GPTJForCausalLM):
-            # For gpt-j model family, this optimization can provide a better performance.
-            model = ipex.optimize(model.eval(), inplace=True)
+
+    model = optimize_model(model, low_bit=low_bit)
+    model = model.eval()
+    if isinstance(model, GPTJForCausalLM):
+        # For gpt-j model family, this optimization can provide a better performance.
+        model = ipex.optimize(model, inplace=True)
+    model = model.to('xpu')
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:443]: need to put this line before the if block?

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 1 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py]: sure

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 3 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py]: sure

**Review by hkvision — APPROVED** [review]

**Pull request merged**